proposals/apple-mdm/Fleet-MDM-Plan.md

```diff
@@ -0,0 +1,218 @@
+
+- List enrollments (the "global" manual enroll and the DEP enroll):
+	`fleetctl apple-mdm enrollments list`
+
+##### DEP enrollment setup
+
+These commands use the <ENROLLMENT_ID>, which is the `enrollments.ID` (obtained via `fleetctl apple-mdm enrollments list`).
+The token downloaded from https://business.apple.com basically links a "DEP enrollment" in Fleet (entry in `apple_dep_enrollments`) to what Apple calls an "MDM server" in https://business.apple.com.
+
+###### 1. `fleetctl apple-mdm enrollments dep set-token setup --enrollment=<ENROLLMENT_ID>`
+
+- Checks the `ENROLLMENT_ID` is a DEP enrollment.
+- Generates `fleet-mdm-apple-dep.pem` and `fleet-mdm-apple-dep.key`:
+	- Stores `fleet-mdm-apple-dep.pem` as a file.
+	- Keeps `fleet-mdm-apple-dep.key` in `~/.fleet/config` under <ENROLLMENT_ID>.
+
+###### 2. Upload PEM to Apple
+
+User uploads `fleet-mdm-apple-dep.pem` to https://business.apple.com, and downloads a `*.p7m` file. Let's call it `fleet-mdm-apple-dep-auth-token-encrypted.p7m`.
+
+###### 3. Finalize DEP setup
+
+`fleetctl apple-mdm enrollments dep set-token finalize --enrollment=<ENROLLMENT_ID> --encrypted-auth-token=fleet-mdm-apple-dep-auth-token-encrypted.p7m`
+	
+- Checks that the `ENROLLMENT_ID` is a DEP enrollment.
+- Decrypts the provided `fleet-mdm-apple-dep-auth-token-encrypted.p7m` with the `fleet-mdm-apple-dep.key` that corresponds to `<ENROLLMENT_ID>` from `~/.fleet/config`.
+- Calls new Fleet API that:
+	- Apply DEP configuration using token (loaded from apple_dep_enrollmentsID==<ENROLLMENT_ID>) and get a "depProfileUUID".
+	- Creates `dep_names` entry with such token + depProfileUUID (as assigner_profile_uuid) (encryption of the token in MySQL will be a TODO.).
+	- Link such `dep_names` entry to the given `<ENROLLMENT_ID>`.
+- If all looks good, then it removes `fleet-mdm-apple-dep.key` from `~/.fleet/config`.
+
+#### Custom commands for DEP enrollments
+
+- List DEP devices: DEP devices of that enrollment and their status (using the DEP proxy API).
+	`fleetctl apple-mdm enrollments dep list <ENROLLMENT_ID>`
+- Sync DEP profiles: makes sure to set the enroll profile for new devices in a DEP enrollment. (Fleet would sync all DEP enrollments automatically every 5m.)
+	`fleetctl apple-mdm enrollments dep sync-profiles <ENROLLMENT_ID>`
+
+#### DEP syncer
+
+Fleet must start a "DEP syncer" routine for each entry in `apple_dep_enrollments` that has a non-NULL `dep_names` (period of 5m).
+
+### Hosts
+
+Initially, we want to account for hosts that are MDM enrolled but not in osquery.
+TODO(Lucas): We'll need to test whether our system and UI support a `hosts` entry w/ an empty `osquery_host_id`...
+
+### MDM commands
+
+These are one-off commands like "DeviceLock", "RestartDevice", etc.
+
+Fleetctl commands:
+- Send command:
+	`fleetctl apple-mdm commands send --target-hosts=1,2,3 --command=foo.plist`
+(fleetctl would be agnostic to contents of "foo.plist")
+- List commands (with their status):
```

Comment:
So this isn't listing all available commands, this is listing commands that have been issued?